feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/widget/SessionsItemDecoration.kt

```diff
@@ -54,34 +65,93 @@ class SessionsItemDecoration(
                 textPaint
             )
         }
     }
 
-    private fun calcTimeText(position: Int, view: View): StartTimeText {
+    private fun calcDateTimeText(
+        position: Int,
+        view: View,
+        isShowDate: Boolean
+    ): StartDateTimeText {
         val sessionItem = adapter.getItem(position) as SessionItem
         val nextSessionItem = if (position < adapter.itemCount - 1) {
             adapter.getItem(position + 1) as SessionItem
         } else null
 
-        var positionY =
-            view.top.coerceAtLeast(sessionTimeTextMarginTopInPx.toInt()) +
-                sessionTimeTextMarginTopInPx + sessionTimeTextSizeInPx
+        // session date text
+        val dateText = if (isShowDate) {
+
+            var sessionDateTextPositionY =
+                view.top.coerceAtLeast(sessionTimeTextMarginTopInPx.toInt()) +
+                        sessionTimeTextMarginTopInPx + sessionTimeTextSizeInPx
+
+            if (sessionItem.startSessionTime() != nextSessionItem?.startSessionTime()) {
+                sessionDateTextPositionY =
+                    sessionDateTextPositionY.coerceAtMost(view.bottom.toFloat())
+            }
+
+            DateText(
+                value = sessionItem.startSessionDate(),
+                positionX = sessionTimeTextMarginStartInPx,
+                positionY = sessionDateTextPositionY.coerceAtMost(
+                    view.bottom.toFloat() -
+                            sessionTimeTextMarginTopInPx -
+                            sessionTimeTextSizeInPx
+                )
+            )
+        } else null
+
+        val sessionTimeTextMarginTop = if (isShowDate) {
+            sessionTimeTextMarginTopInPx + sessionTimeTextSizeInPx + sessionTimeTextMarginTopInPx
+        } else {
+            sessionTimeTextMarginTopInPx
+        }
+
+        // session time text
+        var sessionTimeTextPositionY =
+            (view.top + sessionTimeTextMarginTop.toInt() - sessionTimeTextMarginTopInPx.toInt())
+                .coerceAtLeast(sessionTimeTextMarginTop.toInt()) +
+                    sessionTimeTextMarginTopInPx + sessionTimeTextSizeInPx
+
         if (sessionItem.startSessionTime() != nextSessionItem?.startSessionTime()) {
-            positionY = positionY.coerceAtMost(view.bottom.toFloat())
+            sessionTimeTextPositionY = sessionTimeTextPositionY.coerceAtMost(view.bottom.toFloat())
         }
-        return StartTimeText(
+
+        val startTimeText = StartTimeText(
             value = sessionItem.startSessionTime(),
-            positionX = (sessionTimeSpaceInPx / 2).toFloat(),
-            positionY = positionY
+            positionX = sessionTimeTextMarginStartInPx,
+            positionY = sessionTimeTextPositionY
         )
+
+        return StartDateTimeText(dateText, startTimeText)
     }
 
+    private data class StartDateTimeText(
+        val dateText: DateText?,
+        val startTimeText: StartTimeText
+    )
+
+    private data class DateText(
```

Comment:
Is this almost the same class as `StartTimeText`? then it would be enough to have only one class like `PositionalText`.